sdk/lib/debug/debug.cc

```diff
@@ -101,6 +102,86 @@ namespace
 			write(')');
 		}
 
+		/**
+		 * Helper to print a floating-point value that has already been
+		 * decomposed.
+		 *
+		 * This is used on both the `float` and `double` paths and so is
+		 * factored out.
+		 */
+		void print_float_parts(FloatParts floatParts)
+		{
+			write(static_cast<int32_t>(floatParts.integral));
+			if (floatParts.decimal)
+			{
+				write('.');
+				std::array<char, 10> buf;
+				const char           Digits[] = "0123456789";
+				auto                 s        = floatParts.decimal;
+				for (int i = static_cast<int>(buf.size() - 1); i >= 0; i--)
+				{
+					buf[static_cast<size_t>(i)] = Digits[s % 10];
+					s /= 10;
+				}
+				bool skipZero = true;
+				for (size_t i = buf.size() - floatParts.decimalPlaces;
```

Comment:
Hm.  This and the loop above should run the same number of iterations, but that's not obvious from their spellings... it's probably fine.